src/Commands/Performance.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+namespace CodeIgniter\Tasks\Commands;
+
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\Database\Exceptions\DatabaseException;
+use CodeIgniter\Tasks\Task;
+use CodeIgniter\Tasks\TaskLog;
+
+/**
+ * Shows information on the cache.
+ */
+class Performance extends TaskCommand
+{
+
+	/**
+	 * The Command's name
+	 *
+	 * @var string
+	 */
+	protected $name = 'tasks:performance';
+
+	/**
+	 * the Command's short description
+	 *
+	 * @var string
+	 */
+	protected $description = 'Shows a list of all scheduled tasks with their perforamnce data.';
+
+	/**
+	 * the Command's usage
+	 *
+	 * @var string
+	 */
+	protected $usage = 'tasks:performance';
+
+	/**
+	 * The Command's Options
+	 *
+	 * @var array
+	 */
+	protected $options = [
+		'--task' => 'Get performance data for specific task by alias.',
+		'--n' => 'No of performance records to be displayed for a specific task (must be combined with --task).',
+	];
+
+	/**
+	 * Run The Command
+	 *
+	 * @param array $params
+	 */
+	public function run(array $params)
+	{
+
+		$scheduler = \Config\Services::scheduler();
+
+		config('Tasks')->init($scheduler);
+
+		$tasks = $scheduler->getTasks();
+
+		$tbody = [];
+
+		// If a task option is passed
+
+		if( $taskName = CLI::getOption("task") ){
+			foreach ($tasks as $task)
+			{
+				if( $task->name !== $taskName ){
+					continue;
+				}
+
+				$performanceData = $this->getTaskPerforamnceStats($task);
+
+				$noOfRecords = is_numeric( CLI::getOption("n") ) ? (int) CLI::getOption("n") : 10;
+
+				$performanceRecords = $this->getTaskPerformanceRecords($task, $noOfRecords);
+
+				$tbody = [];
+				$tbody[] = [
+					"Average Execution Duration",
+					$performanceData['avg_duration'] ." s"
+				];
+				$tbody[] = [
+					"Minimum Execution Duration",
+					$performanceData['min_duration'] ." s"
+				];
+				$tbody[] = [
+					"Maximum Execution Duration",
+					$performanceData['max_duration'] ." s"
+				];
+				$tbody[] = [
+					"No of Success Executions",
+					$performanceData['no_of_success']
+				];
+				$tbody[] = [
+					"No of Failure Executions",
+					$performanceData['no_of_failure']
+				];
+
+				/*
+				 * Will be added later
+				$tbody[] = [
+					"Last Run",
+					$task->taskLog->getLastRun()
```

Comment:
Will be implemented in this PR.

I have another ideas in mind, especially for shared hosting limitation for cron jobs to run every 15 minutes ( minimum ).

It can be revisited later.